universal-application-tool-0.0.1/app/filters/DisableCachingFilter.java

```diff
@@ -0,0 +1,32 @@
+package filters;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.concurrent.Executor;
+import javax.inject.Inject;
+import play.mvc.EssentialAction;
+import play.mvc.EssentialFilter;
+
+/** A filter to disable caching our responses. */
+public class DisableCachingFilter extends EssentialFilter {
```

Comment:
question: I know nothing about Play's caching - why do we need a filter do turn off caching? Is it not just a setting somewhere?

Reply:
This is to control the local cache. It puts a few properties in the response header to tell the client not to use a cached version of the page.